veps/sig-network/passt-migration-proposal.md

```diff
@@ -0,0 +1,263 @@
+
+Since virt-launcher drops the root user, this method is secure. However, KubeVirt follows Kubernetes' `Restricted` Pod Security Standard, which disallows CAP_NET_ADMIN.
+
+## Scalability
+
+- `passt` has a memory overhead of ~250Mi per VM. Users running VMs at scale should revert to other bindings if memory overhead is a concern.
+-  VMs with a large number of TCP connections will take longer to live migrate. That's because sockets are handled in a serial
+ fashion, each one separately sent back and forth to passt-repair for handling.
+- The number of concurrent live migration per node is already limited and bound by the number of virt-handler worker goroutines.
+
+## Update/Rollback Compatibility
+
+- Live migration of passt VMs from nodes running older versions of KubeVirt to an upgraded KubeVirt environment will result in TCP connections being reset.
+- Rollback of passt migration functionality (in case system stability is impacted) will be possible by disabling of the feature
+e gate. It is expected that passt would still work, but live migrations would disconnect TCP connections.
+
+
+## Functional Testing
+
+Existing e2e tests cover basic passt connectivity and remain sufficient with vhost-user support. Dedicated e2e tests for seamless migration will verify TCP connection persistence. Sticky IP addresses are required, so tests will use secondary interfaces with hardcoded IPs.
+
+## Implementation Phases
+1. Implement vhost-user in passt-sidecar.
+2. Add passt-repair to virt-handler.
+3. Implement calls to passt-repair in virt-handler.
+4. Enhance e2e tests.
+5. Deployment activities (omitted from this proposal).
+
+Phases can proceed in parallel.
+
+## Feature Lifecycle Phases
+
+`passt` is currently a network binding plugin external to the KubeVirt core, and its use as a network binding is not subject to lifecycle phases. However, the calls to passt-repair from virt-handler impose a risk and should be protected by a feature gate and subject to lifecycle phases.
+
+### Alpha
+
+`passt live migration` will be feature-gated, requiring explicit user opt-in. virt-handler will not run `passt-repair` unless the feature gate is enabled.
+
+passt-sidecar will not condition the feature gate and will populate DomainXML in vhost-user mode regardless of the gate. This part of the code is external to KubeVirt core.
+
+### Beta
+
+For several releases, the live migration process using `passt` network binding, will be optimized. Specifically, `past-repair` and its integration points will be ironed out. 
+The criteria to move to from Alpha to Beta will be stability of the live migration process, and an agreement that the decision to separate `passt-repair` from `passt` was a sustainable one, as other alternatives do exist.
+
+### GA
```

Comment:
GA can be declared when it is part of the core binding and a proper alternative to the masquerade binding.

Reply:
If you can, please clarify that at this point the migration will include seamless connectivity treatment by default with no option to disable it.